mdstcpip/mdsipshr/Connections.c

```diff
@@ -471,7 +471,7 @@ int AddConnection(Connection *c)
   do
   {
     id++; // find next free id
-  } while (id == INVALID_CONNECTION_ID && _FindConnection(id, NULL, MDSIPTHREADSTATIC_VAR));
+  } while ((id == INVALID_CONNECTION_ID) || _FindConnection(id, NULL, MDSIPTHREADSTATIC_VAR));
```

Comment:
Just for tracking, this is fixed re: Timo's comments:
https://github.com/MDSplus/mdsplus/pull/2626#discussion_r1338879619